Modules/_zoneinfo.c

```diff
@@ -1570,21 +1569,24 @@ parse_tz_str(PyObject *tz_str_obj, _tzrule *out)
 }
 
 static int
-parse_uint(const char *const p, uint8_t *value)
+parse_digits(const char **p, int min, int max, int *value)
 {
```

Comment:
Can you add an assertion (or at least a comment) to explain that the loop cannot overflow since max is not greater than 3 (value < 1000)? Otherwise, a reader like me can suspect a risk of overflow (if you don't check call sites).